src/include/storage/tile_group.h

```diff
@@ -76,7 +75,7 @@ class TileGroup : public Printable {
   // Tile group constructor
   TileGroup(BackendType backend_type, TileGroupHeader *tile_group_header,
             AbstractTable *table, const std::vector<catalog::Schema> &schemas,
-            const column_map_type &column_map, int tuple_count);
+            std::shared_ptr<const Layout> layout, int tuple_count);
```

Comment:
We're passing in a shared pointer, does that mean this can be null? If not, should we make it a reference? 

Also, in terms of ownership, does the TileGroup own the layout?

Reply:
The layout is actually shared among multiple TileGroups which are initialized with the same layout. We would want the ptr to be alive as long as we have atleast one TileGroup with the same. 

To make sure a TileGroup always get a non-null layout, I am now throwing an exception in the TileGroupFactory if it passed a `nullptr`. Let me know if you want me to handle this differently.